openmls/src/test_utils/single_group_test_framework/mod.rs

```diff
@@ -0,0 +1,478 @@
+
+        // Initialize the group state
+        let group_id = GroupId::from_slice(b"test");
+        let mut group_state =
+            GroupState::new_from_party(group_id, alice_pre_group, mls_group_create_config).unwrap();
+
+        group_state
+            .add_member(AddMemberConfig {
+                adder: "alice",
+                addees: vec![bob_pre_group, charlie_pre_group],
+                join_config: mls_group_join_config.clone(),
+                tree: None,
+            })
+            .expect("Could not add member");
+
+        group_state.assert_membership();
+
+        group_state
+            .add_member(AddMemberConfig {
+                adder: "bob",
+                addees: vec![dave_pre_group],
+                join_config: mls_group_join_config,
+                tree: None,
+            })
+            .expect("Could not add member");
+
+        group_state.assert_membership();
+    }
+
+    #[openmls_test]
+    pub fn simple_example() {
+        let alice_party = CorePartyState::<Provider>::new("alice");
+        let bob_party = CorePartyState::<Provider>::new("bob");
+
+        let alice_pre_group = alice_party.generate_pre_group(ciphersuite);
+        let bob_pre_group = bob_party.generate_pre_group(ciphersuite);
+
+        // Get the key package for Bob
+        // TODO: should key package be regenerated each time?
+        let bob_key_package = bob_pre_group.key_package_bundle.key_package.clone();
+
+        // Create config
+        let mls_group_create_config = MlsGroupCreateConfig::builder()
+            .ciphersuite(ciphersuite)
+            .use_ratchet_tree_extension(true)
+            .build();
+
+        // Join config
+        let mls_group_join_config = mls_group_create_config.join_config().clone();
+
+        // Initialize the group state
+        let group_id = GroupId::from_slice(b"test");
+        let mut group_state =
+            GroupState::new_from_party(group_id, alice_pre_group, mls_group_create_config).unwrap();
+
+        // Get a mutable reference to Alice's group representation
+        let [alice] = group_state.groups_mut();
+
+        // Build a commit with a single add proposal
+        let bundle = alice
+            .build_commit_and_stage(move |builder| {
+                let add_proposal = Proposal::Add(AddProposal {
+                    key_package: bob_key_package,
+                });
+
+                // ...add more proposals here...
+
+                builder
+                    .consume_proposal_store(false)
+                    .add_proposal(add_proposal)
+            })
+            .expect("Could not stage commit");
+
+        // Deliver_and_apply to all members but Alice
+        group_state
+            .deliver_and_apply_if(bundle.commit().clone().into(), |member| {
+                member.party.core_state.name != "alice"
+            })
+            .expect("Error deliver_and_applying commit");
```

Comment:
I think we should still get alice to apply the locally staged commit

Reply:
Actually at this point the group only contains alice, so this doesn't do anything, or am I missing something?